build-tools/packages/build-infrastructure/src/test/fluidRepo.test.ts

```diff
@@ -104,3 +108,52 @@ describe("loadFluidRepo", () => {
 		});
 	});
 });
+
+describe("setDependencyRange", () => {
+	const repo = loadFluidRepo(testRepoRoot);
+	const main = repo.releaseGroups.get("main" as ReleaseGroupName);
+	assert(main !== undefined);
+
+	const group2 = repo.releaseGroups.get("group2" as ReleaseGroupName);
+	assert(group2 !== undefined);
+
+	const secondWorkspace = repo.workspaces.get("second" as WorkspaceName);
+	assert(secondWorkspace !== undefined);
+
+	afterEach(async () => {
+		await git.checkout(["HEAD", "--", testRepoRoot]);
+		repo.reload();
+	});
+
+	it("updates the dependency range given release group package dependencies", async () => {
+		const group2Version = semver.parse(group2.version);
+		assert(group2Version !== null);
+		await setDependencyRange(main.packages, group2.packages, group2Version);
+		const allCorrect = main.packages
+			.filter((pkg) => group2.packages.some((group2Pkg) => group2Pkg.name === pkg.name))
```

Comment:
Another place to make a Set and use `.has`. Do you think the `packages` properties should be Sets instead of arrays? They aren't supposed to have any duplicates. Maybe the interface should just be defined as an iterable so it can take any iterable collection?

Reply:
I don't this filtering is needed. Release groups are mutually exclusive so main.packages and group2.packages should have no overlap.

Reply:
I realized I was checking the wrong packages in the test, instead of filtering between main.packages and group2.packages, I should be comparing main.packages.dependencies to group2.packages